src/main/java/AuthenticationController.java

```diff
@@ -6,5 +8,26 @@
  * It delegates the request processing to the AuthenticationService to perform the required actions.
  */
 public class AuthenticationController {
-    // TODO implement class
+
+    private AuthenticationService authenticationService;
+
+    public AuthenticationController(AuthenticationService authenticationService) {
+        this.authenticationService = authenticationService;
+    }
+
+    public User createUser(String username, String password) {
+        return authenticationService.createUser(username, password);
+    }
+
+    public boolean signInUser(String username, String password) {
+        return authenticationService.authenticate(username, password);
+    }
+
+    public User changePassword(UUID id, String oldPassword, String newPassword) {
+        return authenticationService.changePassword(id, oldPassword, newPassword);
+    }
+
+    public User changeUsername(UUID id, String currentUsername, String newUsername) {
```

Comment:
You are passing in the `id` here but you are not passing it to the next function. Consider removing it